In [30]:
import numpy as np
import pandas as pd

from statsmodels.formula.api import ols

In [31]:
# import the dataset

sleep_study_data = pd.read_csv('sleep75.csv')

In [32]:
sleep_study_data

,age,black,case,clerical,construc,educ,earns74,gdhlth,inlf,leis1,...,spwrk75,totwrk,union,worknrm,workscnd,exper,yngkid,yrsmarr,hrwage,agesq
0,32,0,1,0.000000,0.000000,12,0,0,1,3529,...,0,3438,0,3438,0,14,0,13,7.070004,1024
1,31,0,2,0.000000,0.000000,14,9500,1,1,2140,...,0,5020,0,5020,0,11,0,0,1.429999,961
2,44,0,3,0.000000,0.000000,17,42500,1,1,4595,...,1,2815,0,2815,0,21,0,0,20.529997,1936
3,30,0,4,0.000000,0.000000,12,42500,1,1,3211,...,1,3786,0,3786,0,12,0,12,9.619998,900
4,64,0,5,0.000000,0.000000,14,2500,1,1,4052,...,1,2580,0,2580,0,44,0,33,2.750000,4096
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
701,45,0,702,0.182331,0.030075,12,5500,1,0,5069,...,1,2026,0,2026,0,27,0,18,NaN,2025
702,34,0,703,0.182331,0.030075,10,2500,0,0,5885,...,0,675,1,465,210,18,0,4,NaN,1156
703,37,0,704,0.182331,0.030075,12,3500,1,0,4719,...,1,1851,0,1851,0,19,0,17,NaN,1369
704,54,0,705,0.182331,0.030075,17,32500,1,0,5149,...,1,1961,1,1481,480,31,0,22,NaN,2916


In [33]:
sleep_study_data.columns.tolist()

['age',
 'black',
 'case',
 'clerical',
 'construc',
 'educ',
 'earns74',
 'gdhlth',
 'inlf',
 'leis1',
 'leis2',
 'leis3',
 'smsa',
 'lhrwage',
 'lothinc',
 'male',
 'marr',
 'prot',
 'rlxall',
 'selfe',
 'sleep',
 'slpnaps',
 'south',
 'spsepay',
 'spwrk75',
 'totwrk',
 'union',
 'worknrm',
 'workscnd',
 'exper',
 'yngkid',
 'yrsmarr',
 'hrwage',
 'agesq']

# **Econometric Problems: t-test**

## 1 t-test:<br>H0: b=0(significance of coefficients)
## *1.1 Problem (sleep equation #1, non robust se)*

For the dataset sleep75, we consider the regression sleep ~ totwrk + age + south + male + smsa + marr + union.<br>
•	Estimate the regression<br>
•	Output the t-test results. Use non-robust (OLS)-se<br>
•	Calculate the critical value<br>
•	Which coefficients are significant?


Round your answer to 3 decimal places. Significance level: 5%.

## **Estimate the regression**

In [34]:
we_need_to_sleeeeeep = ols(formula='sleep~totwrk+age+south+male+smsa+marr+union', data=sleep_study_data).fit()

In [35]:
we_need_to_sleeeeeep.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sleep   R-squared:                       0.131
Model:                            OLS   Adj. R-squared:                  0.123
Method:                 Least Squares   F-statistic:                     15.07
Date:                Sat, 09 May 2026   Prob (F-statistic):           2.11e-18
Time:                        18:03:34   Log-Likelihood:                -5255.9
No. Observations:                 706   AIC:                         1.053e+04
Df Residuals:                     698   BIC:                         1.056e+04
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   3441.2975     78.942     43.593      0.000    3286.305    3596.290
totwrk        -0.1689      0.018     -9.363      0.000      -0.204      -0.133
age            2.8413      1.387      2.048      0.041       0.117       5.565
south        101.6755     41.834      2.430      0.015      19.541     183.810
male          86.0381     34.906      2.465      0.014      17.505     154.571
smsa         -53.9249     33.156     -1.626      0.104    -119.022      11.172
marr          30.0061     41.910      0.716      0.474     -52.278     112.290
union         12.4396     38.096      0.327      0.744     -62.357      87.236
==============================================================================
Omnibus:                       61.619   Durbin-Watson:                   1.946
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              164.554
Skew:                          -0.444   Prob(JB):                     1.85e-36
Kurtosis:                       5.192   Cond. No.                     1.23e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.23e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [36]:
we_need_to_sleeeeeep.params.round(3)

Intercept    3441.297
totwrk         -0.169
age             2.841
south         101.676
male           86.038
smsa          -53.925
marr           30.006
union          12.440
dtype: float64

Regression<br>**sleep** = 3441.297 -0.169\***totwrk** +2.841\***age**+101.676\***south**+86.038\***male**-53.925\***smsa**+30.006\***marr**+12.440\***union**

## **Output the t-test results. Use non-robust (OLS)-se**

In [37]:
from scipy.stats import t # import t-distribution and critical values

In [38]:
# Display the t-statistic for all regressors
# model.t_values
# This method assumes that H0: b=0
we_need_to_sleeeeeep.tvalues

Intercept    43.592692
totwrk       -9.363484
age           2.047834
south         2.430474
male          2.464872
smsa         -1.626416
marr          0.715973
union         0.326534
dtype: float64

In [39]:
# Calculate the t-statistic for one regressor
# our_model.t_test('X=number')
# Our H0: b=0
# Then it is our_model.t_test('X=0')
# If you don't type any number in ('X'), then H0 will be assumed b=0
we_need_to_sleeeeeep.t_test('totwrk=0')

<class 'statsmodels.stats.contrast.ContrastResults'>
                             Test for Constraints                             
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
c0            -0.1689      0.018     -9.363      0.000      -0.204      -0.133

In [40]:
# Manual calculation (less accurate due to rounding numbers previuosly)
# t_stat = (B_estimated - B_h0) / Std_err
t_stat_for_totwrk = (-0.1689 - 0)/0.018
t_stat_for_totwrk

-9.383333333333333

## **Calculate the critical value**

In [41]:
# Option 1
# significance_level (α)= 0.05 = 5%
# degrees_of_freedom = n-k-1 = 706-7-1 (for this specific case)
# n - number of observations
# k - number of regressors
# t.ppf
t.ppf(q=1-0.05/2,df=706-7-1)

np.float64(1.9633684547051278)

In [42]:
# Option 2
# t.isf
t.isf(q=0.05/2,df=706-7-1).round(3)

np.float64(1.963)

#### The critical value is 1.963

#### t_cr (at 5% sign. level) = 1.963

## **Which coefficients are significant?**

### Using t_stat and t_cr (recommended)

Use the **absolute value of the t-statistic** and compare it to the positive critical value. (We consider the case of the Two-sided T-test/Two-Tailed T-test)

If |t_stat| < t_cr => we fail to reject the null hypothesis (H0) => the X is **not correlated** with Y in the population (at α significance level)

If |t_stat| > t_cr => we reject the null hypothesis (H0) => the X is **correlated** with Y in the population (at α sign level)

•	totwrk t_stat = -9.363<br>
•	totwrk t_stat module = |-9.363| = 9.363<br>
•	totwrk t_cr = 1.963<br>
Answer: If **totwrk** |t_stat| > t_cr (|-9.363| > 1.963) => we reject the null hypothesis (H0) => the **totwrk** is correlated with **sleep** in the population (at **5%** significance level).

### Another way:<br>Compare to the p-value


If the p-value > alpha/α/significance level => we fail to reject the null (H0) => the X is **not correlated** with Y in the population (at α sign level)

If the p-value < alpha/α/significance level => we reject the null hypothesis (H0) => the X is **correlated** with Y in the population (at α sign level)

Answer: If **totwrk** P-value < α (0.000 < 0.05) => we reject the null hypothesis (H0) => the **totwrk** is correlated with **sleep** in the population (at **5%** significance level).